ui/src/app/cohort-search/attributes-page-v2/attributes-page-v2.component.tsx

```diff
@@ -522,18 +614,102 @@ export const AttributesPageV2 = fp.flow(withCurrentWorkspace(), withCurrentCohor
+          <FlexRowWrap>
+            {![null, 'ANY'].includes(attr.operator) && <div style={{width: '33%'}}>
+              <NumberInput style={{padding: '0 0.25rem', ...(this.hasUnits ? {width: '70%'} : {})}}
+                           value={attr.operands[0] || ''}
+                           min={attr.MIN} max={attr.MAX}
+                           onChange={(v) => this.inputChange(v, a, 0)}/>
+              {this.hasUnits && <span> {PM_UNITS[subtype]}</span>}
+            </div>}
+            {attr.operator === Operator.BETWEEN && <React.Fragment>
+              <div style={{padding: '0.2rem 1.5rem 0 1rem'}}>and</div>
+              <div style={{width: '33%'}}>
+                <NumberInput style={{padding: '0 0.25rem', ...(this.hasUnits ? {width: '70%'} : {})}}
+                             value={attr.operands[1] || ''}
+                             min={attr.MIN} max={attr.MAX}
+                             onChange={(v) => this.inputChange(v, a, 1)}/>
+                {this.hasUnits && <span> {PM_UNITS[subtype]}</span>}
+              </div>
+            </React.Fragment>}
+          </FlexRowWrap>
+          {this.hasRange && ![null, 'ANY'].includes(attr.operator) && <div style={{paddingTop: '0.2rem'}}>
+            Range: {attr.MIN.toLocaleString()} - {attr.MAX.toLocaleString()}
+          </div>}
+        </div>)}
+      </React.Fragment>;
+    }
+
+    renderCategoricalAttributes() {
+      const {node: {count}} = this.props;
+      const {form, isCOPESurvey} = this.state;
+      return form.cat.length > 0 && <React.Fragment>
+        {isCOPESurvey && <div>
+          <CheckBox onChange={(v) => this.toggleAnyVersionCheckbox(v)}/> Any version
+          {count > -1 && <span style={styles.badge}>{count.toLocaleString()}</span>}
+        </div>}
+        {!(isCOPESurvey && form.anyVersion) && <React.Fragment>
+          <div style={styles.orCircle}>OR</div>
+          {!isCOPESurvey && <div style={styles.label}>Categorical Values</div>}
+          {form.cat.map((attr, a) => <div key={a} style={styles.categorical}>
+            <CheckBox checked={attr.checked} style={{marginRight: '3px'}}
+                      onChange={(v) => this.checkboxChange(v, a)}/>
+            {attr.conceptName}
+            <span style={styles.badge}>{parseInt(attr.estCount, 10).toLocaleString()}</span>
+          </div>)}
+        </React.Fragment>}
+      </React.Fragment>;
+    }
+
     render() {
       const {close, node: {domainId, name, parentId, subtype}} = this.props;
-      const {calculating, count, countError, form, loading, options} = this.state;
+      const {calculating, count, countError, form, isCOPESurvey, loading} = this.state;
       const {formErrors, formValid} = this.validateForm();
       const disableAdd = calculating || !formValid;
-      const disableCalculate = disableAdd || form.exists || form.num.every(attr => attr.operator === 'ANY');
+      const disableCalculate = disableAdd
+        || (form.anyValue && count !== null)
```

Comment:
Can you move this to a method just so its easier to read?

Reply:
Done. I also did a small refactor to move `formErrors` and `formValid` into state.